src/util/glob.ts

```diff
@@ -190,14 +205,20 @@ export async function glob(
   // Pass each group to fast-glob to match in parallel, and combine into a
   // single set.
   const combinedMap = new Map<string, Entry>();
+  // Ensure the cwd is absolute and normalized so that we can do path string
+  // comparisons.
+  const normalizedCwd = pathlib.resolve(opts.cwd);
+  const normalizedCwdWithTrailingSep = normalizedCwd + pathlib.sep;
   await Promise.all(
     groups.map(async ({include, exclude}) => {
       const matches = await fastGlob(include, {
         ignore: exclude,
-        cwd: opts.cwd,
+        cwd: normalizedCwd,
         dot: true,
         onlyFiles: !opts.includeDirectories,
-        absolute: opts.absolute,
+        // Return absolute paths, even if we ultimately return relative ones, so
+        // that we can do path string comparisons.
+        absolute: true,
```

Comment:
Is the path string comparison only necessary if `opts.throwIfOutsideCwd` is true? I wonder if it would be much more complex to only override the `absolute` passed to `fastGlob` and revert to relative paths if needed, only if `throwIfOutsideCwd` is true.